libs/datamodel/connectors/sql-datamodel-connector/src/mssql_datamodel_connector.rs

```diff
@@ -32,31 +75,163 @@ impl Connector for MsSqlDatamodelConnector {
     }
 
-    fn validate_model(&self, _model: &Model) -> Result<(), ConnectorError> {
+    fn validate_model(&self, model: &Model) -> Result<(), ConnectorError> {
+        for index_definition in model.indices.iter() {
+            let fields = index_definition.fields.iter().map(|f| model.find_field(f).unwrap());
+
+            for field in fields {
+                if let FieldType::NativeType(_, native_type) = field.field_type() {
+                    // We've validated the type earlier (hopefully).
+                    let r#type = MsSqlType::try_from(native_type).unwrap();
+
+                    if MsSqlType::heap_allocated().contains(&r#type) {
+                        if index_definition.tpe == IndexType::Unique {
+                            return Err(ConnectorError::new_incompatible_native_type_with_unique(
+                                &format!("{}", r#type),
+                                "SQL Server",
+                            ));
+                        } else {
+                            return Err(ConnectorError::new_incompatible_native_type_with_index(
+                                &format!("{}", r#type),
+                                "SQL Server",
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+
+        for id_field in model.id_fields.iter() {
+            let field = model.find_field(id_field).unwrap();
+
+            if let FieldType::NativeType(_, native_type) = field.field_type() {
+                // We've validated the type earlier (hopefully).
+                let r#type = MsSqlType::try_from(native_type).unwrap();
+
+                if MsSqlType::heap_allocated().contains(&r#type) {
+                    return Err(ConnectorError::new_incompatible_native_type_with_id(
+                        &format!("{}", r#type),
+                        "SQL Server",
+                    ));
+                }
+            }
+        }
+
         Ok(())
     }
 
     fn available_native_type_constructors(&self) -> &Vec<NativeTypeConstructor> {
         &self.constructors
     }
 
-    fn parse_native_type(&self, _name: &str, _args: Vec<String>) -> Result<NativeTypeInstance, ConnectorError> {
-        Err(ConnectorError::from_kind(
-            ErrorKind::ConnectorNotSupportedForNativeTypes {
-                connector_name: "mssql".to_string(),
-            },
+    fn parse_native_type(&self, name: &str, args: Vec<String>) -> Result<NativeTypeInstance, ConnectorError> {
+        let qualified = if args.len() > 0 {
+            Cow::from(format!("{}({})", name, args.join(",")))
+        } else {
+            Cow::from(name)
+        };
+
+        // Unwrapping as the core must guarantee to just call with known names.
+        let native_type: MsSqlType = qualified.parse().unwrap();
+
+        Ok(NativeTypeInstance::new(
+            native_type.kind(),
+            native_type.parameters(),
+            &native_type,
         ))
     }
```

Comment:
This function does not return any parsing errors compared to the other connector implementations. Are we sure that this is not needed? If yes we should take a note to to adapt the other implementations.

Reply:
This is not good. Fixing. Wasn't sure at all where the actual validation should happen. It seems now it's in this function.

Reply:
It might be that enough validation is happening in the core. It should just be checked to be sure.